pkcs11/signer.go

```diff
@@ -91,18 +109,19 @@ func getSigner(ctx context.Context, requestChan chan scheduler.Request, pool sPo
 		// This should ideally not happen but in order to avoid a blocking call we add this check in place.
 		return nil, errors.New("request channel is closed, cannot fetch signer")
 	}
-	var ok bool
+
 	select {
-	case signer, ok = <-respChan:
-		if signer == nil || !ok {
-			return nil, errors.New("client request timed out, skip signing cert request")
-		}
+	case err := <-errChan:
+		return nil, err
+	case resp := <-respChan:
+		return resp, nil
 	case <-ctx.Done():
-		// In order to ensure we don't keep on blocking on the response, we close the response channel for this request & return.
+		// Closing respChan and errChan guarantees the goroutine gets panic() when writing data on the other side.
 		close(respChan)
+		close(errChan)
```

Comment:
acc. to golang standard it should be the sender who closes the channel & not the receiver as receiver does not know when the sender is sending the data resulting in panic. ref: [channel-closing](https://go101.org/article/channel-closing.html#:~:text=One%20general%20principle%20of%20using,only%20sender%20of%20the%20channel.)